bookOfHb.md

```diff
@@ -0,0 +1,400 @@
+# The Book of Self-Repair (Heartbeat First Impressions)
+
+##  Overall Goals of Self-Repair
+
+1.  Host self-repair recipes, outside of firefox core.  Run them with Firefox permissions.  Allow fast development off the trains.
+
+2.  Collect randomly sampled User Sentiment scores.  Implemented as "Heartbeat First Impressions" recipe.
+
+
+## As a developer, I want to successfully hack on Self-Repair
+
+### Setup (do once)
+
+1.  Zen Of Cooking
+
+	This is the short version. `Cmd+F` will find bigger explanations of nearly everything later on in this document.  
+
+   Mastering this may take several attempts.  Skip around.  Be gentle on yourself!  Enjoy the process!
+   
+1. Pots and Pans
+
+	```
+	# required: git, node
+	brew install node   # includes npm
+	brew install git    # git
+	
+	# optional:  hub, git
+	brew install hub    # github's replacement git client
+	brew install ghi    # github issues client
+	```
+
+1. Groceries
+
+	```
+	# usual git project clone and fork
+	
+	git clone mozilla/self-repair-server
+	cd self-repair-server
+	hub fork
+	```
+
+1. Prep the veggies
+
+	```
+	# usual node project package install
+	npm install   
```

Comment:
After tweaking package.json, I installed everything, but because thirdparty isn't installed, I can't do `npm run compile`.

Seems like there's a minimum node.js and npm required and that should be specified in the Pots and Pans section.